storage/libradb/src/pruner/mod.rs

```diff
@@ -220,6 +230,35 @@ impl Worker {
             }
         }
     }
+
+    /// Purge retired records so that after restart not too much already pruned stuff is dealt with
+    /// again (although no harm is done deleting those then non-existent things.)
+    ///
+    /// We issue the deletes only periodically instead of after every pruning batch to avoid sending
+    /// too many deletions to the DB, which takes disk space and slows it down.
+    fn maybe_purge_retired_record(&mut self) -> Result<()> {
+        const MIN_INTERVAL: Duration = Duration::from_secs(60);
+        const MIN_VERSIONS: u64 = 60000;
+
+        // A deletion is issue at most once in one minute and when the pruner progressed by at
+        // least 60000 versions (assuming the pruner deletes as slow as 1000 versions per second
+        // this imposes at most one minute of work in vain after restarting.)
+        let now = Instant::now();
+        if now - self.retired_record_purged_at > MIN_INTERVAL {
+            let progress = self.progress.load(Ordering::Relaxed);
+
+            if progress - self.retired_record_purged_till > MIN_VERSIONS {
+                self.db.range_delete::<RetiredStateRecordSchema, Version>(
```

Comment:
Are you also deleting these in `prune_state` right now?

Reply:
good catch.. at some point I decided to `git reset --hard` :p

updated